packages/api/src/orbitAPISchema.json

```diff
@@ -0,0 +1,527 @@
+                                                            "timestampMillis",
+                                                            "updates"
+                                                        ],
+                                                        "type": "object"
+                                                    }
+                                                ]
+                                            },
+                                            "id": {
+                                                "additionalProperties": false,
+                                                "properties": {
+                                                },
+                                                "type": "object"
+                                            },
+                                            "objectType": {
+                                                "enum": [
+                                                    "actionLog"
+                                                ],
+                                                "type": "string"
+                                            }
+                                        },
+                                        "required": [
+                                            "data",
+                                            "id",
+                                            "objectType"
+                                        ],
+                                        "type": "object"
+                                    },
+                                    "type": "array"
+                                },
+                                "hasMore": {
+                                    "type": "boolean"
+                                },
+                                "objectType": {
+                                    "enum": [
+                                        "list"
+                                    ],
+                                    "type": "string"
+                                }
+                            },
+                            "required": [
+                                "data",
+                                "hasMore",
+                                "objectType"
+                            ],
+                            "type": "object"
+                        }
+                    },
+                    "required": [
+                        "query"
+                    ],
+                    "type": "object"
+                },
+                "PATCH": {
+                    "additionalProperties": false,
+                    "properties": {
+                        "body": {
+                            "items": {
+                                "additionalProperties": false,
+                                "properties": {
+                                    "data": {
+                                        "anyOf": [
```

Comment:
It's a shame that the definition for `ActionLog` ends up repeated here. It's probably not that big a deal, but if we do end up validating API responses in `app`, we'll be shipping this JSON file to browsers. I don't suppose there's a flag we're missing that can indirect types? If not I imagine we may skip validating responses client-side.

Reply:
Ah yes, I noticed this as well. For this to work, we could enable the `aliasRefs` flag. From their docs:
```
  --aliasRefs           Create shared ref definitions for the type aliases.          [boolean] [default: false]
```
However, this flag seems to create a lot of noise since every type will now be defined and it appears to be broken for generics. Instead, I converted the `ActionLog` to be defined using `interface`s rather than `type` which indicates to export the type as a shared ref. This ends up reducing the lint count quite a bit!

Example:
```json
"data": {
  "anyOf": [
      {
          "$ref": "#/definitions/IngestActionLog"
      },
      {
          "$ref": "#/definitions/RepetitionActionLog"
      },
      {
          "$ref": "#/definitions/RescheduleActionLog"
      },
      {
          "$ref": "#/definitions/UpdateMetadataActionLog"
      }
  ]
},
```

Reply:
Hilariously, I remember switching `ActionLog` from an `interface` to a `type` a while back because TypeScript couldn't infer something I expected it to. Alas, I don't remember the specifics. I think the issue is that `interface`s are "open"—i.e. subtypes can add fields—and this broke some exhaustiveness guarantee I needed. Assuming that the whole project still builds, this should be fine!